CONTRIBUTING.md

```diff
@@ -0,0 +1,91 @@
+- [ ] Your fork is up-to-date with the source repository (`peterhuene/azure-functions-rs`)
+- [ ] All tests pass.
+- [ ] "TODO" comments are removed.
+- [ ] Temp variables are given good names.
+- [ ] Any merge conflicts have the potential to be reasonably sorted out.
+
+If you are uncertain about your contribution, that's ok! You can always make a [draft pull request](https://help.github.com/en/articles/about-pull-requests#draft-pull-requests).
+
+## Repository Layout
+
+This repository is split into multiple Rust crates:
+
+- [azure-functions](https://github.com/peterhuene/azure-functions-rs/tree/master/azure-functions) - The `azure-functions` crate that defines the types and functions that are used when writing Azure Functions with Rust.
+- [azure-functions-codegen](https://github.com/peterhuene/azure-functions-rs/tree/master/azure-functions-codegen) - The `azure-functions-codegen` crate that defines the procedural macros that are used when writing Azure Functions with Rust.
+- [azure-functions-sdk](https://github.com/peterhuene/azure-functions-rs/tree/master/azure-functions-sdk) - The `azure-functions-sdk` crate that implements the `cargo func` command.
+- [azure-functions-shared](https://github.com/peterhuene/azure-functions-rs/tree/master/azure-functions-shared) - The `azure-functions-shared` crate that defines types and functions that are shared between the `azure-functions-codegen` and `azure-functions` crates.
+  - Note: the `azure-functions-shared/protobuf` directory is the git submodule for [Azure Functions Language Worker Protocol](https://github.com/Azure/azure-functions-language-worker-protobuf).
+- [azure-functions-shared-codegen](https://github.com/peterhuene/azure-functions-rs/tree/master/azure-functions-shared-codegen) - The `azure-functions-shared-codegen` crate that defines the procedural macros used by the shared `azure-functions-shared` crate.
+- [examples](https://github.com/peterhuene/azure-functions-rs/tree/master/examples) - The directory containing example Azure Functions.
+
+## Setting Up a Dev Environment
+
+### Cloning the Repository
+
+This repository uses a git submodule for defining the [Azure Functions Language Worker Protocol](https://github.com/Azure/azure-functions-language-worker-protobuf).
+
+Use `--recurse-submodules` when cloning this repository:
+
+Cloning with SSH:
+
+``` bash
+git clone --recurse-submodules git@github.com:<GITHUB-USERNAME>/azure-functions-rs.git
+```
+
+Cloning with HTTPS:
+
+``` bash
+git clone --recurse-submodules https://github.com/<GITHUB-USERNAME>/azure-functions-rs.git
+```
+
+If you want to clone the source repository, replace `<GITHUB-USERNAME>` with `peterhuene`. To clone your own fork, replace the value with your GitHub username.
+
+### Building
+
+Build at the root of the repository to build both the `azure-functions-codegen` and the `azure-functions` libraries using `cargo build`:
+
+``` bash
+cargo build
+```
+
+### Running tests
```

Comment:
Can we insert a new section here that describes how to install the Azure Functions SDK from source?

The command (from the root) would be:

```
cargo install -f --path azure-functions-sdk
```

Reply:
I added the section. I wasn't sure what to say - if there is a motivating statement as to why installing from source is helpful. What I wrote was simply:

### Installing Azure Functions SDK from Source

To install the Azure Functions SDK from source, from the root of the project run  

``` bash
cargo install -f --path azure-functions-sdk
```